### PR TITLE
Enable async option for spring-cloud library

### LINKED ISSUE #2327: [BUG] [Spring] Async missing import
When specifying
```html
<configOptions>
	<java8>true</java8>
	<dateLibrary>java8</dateLibrary>
	<library>spring-cloud</library>
	<async>true</async>
</configOptions>
<generatorName>spring</generatorName>
```
the generated API Interface file is missing the import for `CompletableFuture`
(assumably `import java.util.concurrent.CompletableFuture;`)

### PR DESCRIPTION
**Pull request by Zomzog**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Enable async option for spring-cloud library

Fix #2327

@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03)

### CODE REVIEW

**Comment by Zomzog** [comment]
Change the PR for comment on #2327

File: samples/client/petstore/spring-cloud-async/src/main/java/org/openapitools/api/PetApi.java [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+            @AuthorizationScope(scope = "read:pets", description = "read your pets")
+            })
+    }, tags={ "pet", })
+    @ApiResponses(value = { 
+        @ApiResponse(code = 405, message = "Invalid input") })
+    @RequestMapping(value = "/pet",
+        consumes = "application/json",
+        method = RequestMethod.POST)
+    default CompletableFuture<ResponseEntity<Void>> addPet(@ApiParam(value = "Pet object that needs to be added to the store" ,required=true )  @Valid @RequestBody Pet body) {
+        return CompletableFuture.completedFuture(new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED));
```
Comment (cbornet) [inline samples/client/petstore/spring-cloud-async/src/main/java/org/openapitools/api/PetApi.java]: Feign interfaces must not have default implementation so I don't think this works.

Comment (Zomzog) [reply samples/client/petstore/spring-cloud-async/src/main/java/org/openapitools/api/PetApi.java]: You're right. I have redone the PR code for it.

**Comment by wing328** [comment]
@Zomzog when you've time, can you please resolve the merge conflicts?

**Comment by Zomzog** [comment]
@wing328 Rebase done.

**Comment by wing328** [comment]
@Zomzog can you please resolve the merge conflicts again? I'm sorry for the delay in reviewing your PR which results in merge conflicts.

**Comment by Zomzog** [comment]
@wing328 merge conflicts resolved.

**Comment by wing328** [comment]
@Zomzog thanks for the PR, which has been included in the 4.1.0 release: https://twitter.com/oas_generator/status/1160000504455319553

**Pull request merged**